petab/core.py

```diff
@@ -595,3 +636,106 @@ def create_measurement_df():
     })
 
     return df
+
+
+def create_parameter_df(sbml_model, condition_df, measurement_df,
+                        parameter_scale='log10',
+                        lower_bound=None,
+                        upper_bound=None):
+    """Create a new PEtab parameter table
+
+    All table entries can be provided as string or list-like with length
+    matching the number of parameters
+
+    Arguments:
+        sbml_model: @type libsbml.Model
+        condition_df: @type pandas.DataFrame
+        measurement_df: @type pandas.DataFrame
+        parameter_scale: parameter scaling
+        lower_bound: lower bound for parameter value
+        upper_bound: upper bound for parameter value
+    """
+
+    # We do not handle that yet, once we do, we need to add those parameters
+    # as well
+    lint.condition_table_is_parameter_free(condition_df)
+
+    observables = get_observables(sbml_model)
+    sigmas = get_sigmas(sbml_model)
+
+    # collect placeholder parameters
+    placeholders = set()
+    for k, v in observables.items():
+        placeholders |= get_placeholders(v['formula'], get_observable_id(k),
+                                         'observable')
+    for k, v in sigmas.items():
+        placeholders |= get_placeholders(v, get_observable_id(k), 'noise')
+
+    # collect assignment targets
+    observables = observables.keys()
+    sigmas = ['sigma_' + get_observable_id(k) for k in sigmas.keys()]
+
+    # grab all from model and measurement table
+    # without condition table parameters
+    # and observables assigment targets
+    # and sigma assignment targets
+    # and placeholder parameters (only partial overrides are not supported)
+    # use ordered dict as proxy for ordered set
+    parameter_ids = OrderedDict.fromkeys(
+        p.getId() for p in sbml_model.getListOfParameters()
+        if p.getId() not in condition_df.columns
+        and p.getId() not in observables
+        and p.getId() not in sigmas
+        and p.getId() not in placeholders)
+
+    # Append parameters from measurement table,
+    # unless they are fixed parameters
+    def append_overrides(overrides):
+        for p in overrides:
+            if isinstance(p, str) and p not in condition_df.columns:
+                parameter_ids[p] = None
+    for _, row in measurement_df.iterrows():
+        # we trust that the number of overrides matches
+        append_overrides(
+            split_parameter_replacement_list(row.observableParameters))
+        append_overrides(
+            split_parameter_replacement_list(row.noiseParameters))
+
+    parameter_ids = list(parameter_ids.keys())
+
+    if lower_bound is None:
+        lower_bound = [-3] * len(parameter_ids)
+
+    if upper_bound is None:
+        upper_bound = [-5] * len(parameter_ids)
```

Comment:
They are super arbitrary, and often not what you want. Alternatives are inf or nan... Maybe I will go for the latter, since the default doesn't safe too much typing anyways...

Reply:
I'd favorize the latter (and maybe add parameters to the function for the user to specify them), since this will probably reduce wrong usage. And if optimizers cannot handle inf bounds, they will notify the user.